images/image.go

```diff
@@ -119,7 +119,7 @@ func (image *Image) Size(ctx context.Context, provider content.Provider, platfor
 		}
 		size += desc.Size
 		return nil, nil
-	}), FilterPlatforms(ChildrenHandler(provider), platform)), image.Target)
+	}), LimitManifests(FilterPlatforms(ChildrenHandler(provider), platform), platform, 1)), image.Target)
```

Comment:
This change looks reasonable for me. But we need to know the other use cases, for example, someone uses `platforms.All` to get the sum of all the platforms. I am not sure this case is reasonable because we cannot use arm platform image content in linux/amd64 platform.

WDYT? @dmcgowan @stevvooe

Reply:
I think maybe we need to distinguish here between image Size and Usage. Perhaps usage is what is really concerned about here and we should just be skipping any resource which cannot be found locally. If that is the case, if we just have a `Size` function which can determine the size of an image given a platform, that would error out if it cannot find the manifests, however I am not sure how useful that function is for anything except introspection. Likewise a usage function may want to account for snapshot usage as well as blob usage.